neuralmonkey/config/configuration.py

```diff
@@ -67,11 +63,40 @@ def load_file(self, path):
             log("Failed to load INI file: {}".format(exc), color='red')
             traceback.print_exc()
             exit(1)
-        finally:
-            config_f.close()
 
         return arguments
 
+    def build_model(self):
+        log("Building model based on the config.")
+        try:
+            model = build_config(self.config_dict, self.ignored)
+        #pylint: disable=broad-except
+        except Exception as exc:
+            log("Failed to build model: {}".format(exc), color='red')
+            traceback.print_exc()
+            exit(1)
+        log("Model built.")
+        self._check_loaded_conf(model)
+        model_n = Namespace()
+        for name, value in model.items():
+            if name in self.conditions and not self.conditions[name](value):
+                cond_code = self.conditions[name].__code__
+                cond_filename = cond_code.co_filename
+                cond_line_number = cond_code.co_firstlineno
+                raise Exception(
+                    "Value of field '{}' does not satisfy "
+                    "condition defined at {}:{}."
+                    .format(name, cond_filename, cond_line_number))
+
+            setattr(model_n, name, value)
+            #arguments.__dict__[name] = value
+
+        for name, value in self.defaults.items():
+            if name not in model_n.__dict__:
+                model_n.__dict__[name] = value
```

Comment:
Hlavní problém je v tom, že nevím proč to tam vůbec bylo, takže se mi teď těžko rozhoduje, kde (pokud vůbec) to bude chybět...